src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py

```diff
@@ -200,6 +242,94 @@ def get_max_surge(self):
         return max_surge
 
+    # pylint: disable=too-many-branches
+    def get_node_count_and_enable_cluster_autoscaler_min_max_count(
+        self,
+    ) -> Tuple[int, bool, Union[int, None], Union[int, None]]:
+        """Obtain the value of node_count, enable_cluster_autoscaler, min_count and max_count.
+
+        This function will verify the parameters through function "__validate_counts_in_autoscaler" by default.
+
+        :return: a tuple containing four elements: node_count of int type, enable_cluster_autoscaler of bool type,
+        min_count of int type or None and max_count of int type or None
+        """
+        # node_count
+        # read the original value passed by the command
+        node_count = self.raw_param.get("node_count")
+        # try to read the property value corresponding to the parameter from the `agentpool` object
+        if self.agentpool and self.agentpool.count is not None:
+            node_count = self.agentpool.count
+
+        # enable_cluster_autoscaler
+        # read the original value passed by the command
+        enable_cluster_autoscaler = self.raw_param.get("enable_cluster_autoscaler")
+        # try to read the property value corresponding to the parameter from the `agentpool` object
+        if self.agentpool and self.agentpool.enable_auto_scaling is not None:
+            enable_cluster_autoscaler = self.agentpool.enable_auto_scaling
+
+        # min_count
+        # read the original value passed by the command
+        min_count = self.raw_param.get("min_count")
+        # try to read the property value corresponding to the parameter from the `agentpool` object
+        if self.agentpool and self.agentpool.min_count is not None:
+            min_count = self.agentpool.min_count
+
+        # max_count
+        # read the original value passed by the command
+        max_count = self.raw_param.get("max_count")
+        # try to read the property value corresponding to the parameter from the `agentpool` object
+        if self.agentpool and self.agentpool.max_count is not None:
+            max_count = self.agentpool.max_count
+
+        # these parameters do not need dynamic completion
+
+        # validation
+        self.__validate_counts_in_autoscaler(
+            node_count,
+            enable_cluster_autoscaler,
+            min_count,
+            max_count,
+            decorator_mode=DecoratorMode.CREATE,
+        )
+        return node_count, enable_cluster_autoscaler, min_count, max_count
+
+    def get_node_osdisk_size(self) -> Union[int, None]:
+        """Obtain the value of node_osdisk_size.
+
+        Note: SDK performs the following validation {'maximum': 2048, 'minimum': 0}.
+
+        This function will normalize the parameter by default. The parameter will be converted to int.
+
+        :return: int or None
+        """
+        # read the original value passed by the command
+        node_osdisk_size = self.raw_param.get("node_osdisk_size")
```

Comment:
is there any place to check the format of the string?

Reply:
Fixed in commit [#b9cf823](https://github.com/Azure/azure-cli/pull/21598/commits/b9cf8238fac98d5bb51e3a0f6b74a4fca8247438), added type restrictions to parameter declarations, cli will properly handle type errors.

Also fixed incorrect parameter declaration for `node_osdisk_type`.